### ISSUE TITLE
hardware token pkcs11.exceptions.NoSuchKey after upgrading to 0.23.0

### ISSUE THREAD

**Issue by C-monC** [MatthiasValvekens/pyHanko#394]
**Describe the bug**
Using a hardware token stopped working in the recent release of PyHanko.

```
lib = pkcs11.lib("/usr/lib/libeTPkcs11.so")
token = lib.get_token(token_label="my_token")
with token.open(user_pin="*****") as session:
    for obj in session.get_objects({
        Attribute.CLASS: ObjectClass.PRIVATE_KEY,
    }):
        key_id = obj[Attribute.ID]
        pkcs11_conf = PKCS11SignatureConfig(
            module_path="/usr/lib/libeTPkcs11.so",
            cert_label="tq-a43cf071******1b10724",
            key_id=key_id,
            user_pin="******",
            prompt_pin=PKCS11PinEntryMode.PROMPT,
            token_criteria=TokenCriteria(
                label="my_token", serial=b"027**db"
            ),
        )
with PKCS11SigningContext(pkcs11_conf) as signer:
                    pdf_signer = signers.PdfSigner(  <--------------------------------------- Exception on this line 
                        signature_meta,
                        timestamper=timestamper,
                        signer=signer,  # type: ignore
                        stamp_style=stamp.StaticStampStyle.from_pdf_file(
                            stamp_pdf,
                            border_width=0,
                        ),
                    )
```
The above code works with pyHanko==0.21.0 and not with pyHanko==0.23.0. I haven't worked through the recent changes yet but I'll get on it soon.

Exception thrown: pkcs11.exceptions.NoSuchKey: No key matching {&lt;Attribute.CLASS&gt;: &lt;ObjectClass.PRIVATE_KEY&gt;, &lt;Attribute.LABEL&gt;: 'tq-a43cf071-*****2ad1b10724', &lt;Attribute.ID&gt;: b'\x99*****xa3'}

**Comment by MatthiasValvekens** [comment]
Hi @C-monC, this was a regression introduced in `0.22.0`. I've committed a fix, will do a patch release soon.

**Comment by MatthiasValvekens** [comment]
Released yesterday as 0.23.1.

**Comment by C-monC** [comment]
Just confirming it is fixed. Thanks for the speedy patch :)
